drivers/soundwire/stream.c

```diff
@@ -136,8 +136,12 @@ static int sdw_program_slave_port_params(struct sdw_bus *bus,
 	dpn_prop = sdw_get_slave_dpn_prop(s_rt->slave,
 					  s_rt->direction,
 					  t_params->port_num);
-	if (!dpn_prop)
-		return -EINVAL;
+	if (!dpn_prop) {
```

Comment:
@RanderWang Could you add a comment to describe why dpn_prop is null is not an error here?

Reply:
Good analysis @RanderWang but we should maybe check what the stream status to decide if that port needs to be programmed first?

Removing an error case doesn't seem quite right.

Reply:
@plbossart  it is not feasible to check stream status.  we first program port in _sdw_prepare_stream at https://github.com/thesofproject/linux/blob/integration/soundwire-intel/drivers/soundwire/stream.c#L1494,  then stream->state is set to SDW_STREAM_PREPARED at https://github.com/thesofproject/linux/blob/integration/soundwire-intel/drivers/soundwire/stream.c#L1534. So When setting port params, the stream status may be SDW_STREAM_CONFIGURED, not SDW_STREAM_PREPARED.